### DISCUSSION TITLE
如何强制全站开启HTTPS

### DISCUSSION THREAD

**Discussion by ichswarox** [natfrp/wiki#461] [使用问题] [answered]
设置好SSL后，访问Wordpress网站提示mixed content，有链接使用了http加载导致了页面加载问题，Nginx Proxy Manager有Force Https选项，请问Natfrp如何启用？
<img width="848" alt="image" src="https://github.com/natfrp/wiki/assets/18469758/6d53334e-4ce4-42e2-8d15-93281af937c5">

**Comment by fakeboboliu** [comment, marked as answer]
此问题以通常思路来说，为您的程序（Wordpress 或其主题）设置问题，请将为您的程序正确设置 base url 或者使用 protocol-relative link。

我们不推荐让我们的程序来画蛇添足，实为没有必要的举动。

如果您需要我们来为您做这件事，请打开创建 http 重定向并调整到 301，这与 NPM 的 [Force SSL 开关](https://github.com/NginxProxyManager/nginx-proxy-manager/blob/develop/docker/rootfs/etc/nginx/conf.d/include/force-ssl.conf) 是等价的。

**Comment by ichswarox** [comment]
好的，已经开启了，目前看问题还未解决，关于未在程序中设置base url的原因是因为部署方式使用的是docker，所以这个base url置灰，无法调整，尝试使用了插件，但是仍然不能解决问题，依然会有mix content问题。请问下我的设置方式正确吗，万分感谢。
![image](https://github.com/natfrp/wiki/assets/18469758/0ef87e50-79ae-47e5-ae6e-b0483f67079d)

**Reply by fakeboboliu** [reply]
设置后行为与您提到的 NPM 等价，如果您还有问题，请检查自己的程序。

通常来讲，解决方案是设置 base url，这就不属于我们能管的了。
